pub/scripts/ChartLegend.js

```diff
@@ -50,57 +50,107 @@ ChartLegend.prototype.bindLegendInteractions = function () {
       this.controller.updatePageState()
     }.bind(this))
 
+    // open color input
+    series.legendEl.find('.legend-color').click(function (event) {
+      event.stopPropagation()
+      this.removePopovers()
+      this.openColorInput(series)
+    }.bind(this))
+
     // open move-chart popover
     series.legendEl.find('.move-chart').click(function (event) {
       event.stopPropagation()
-      this.$container.find('.move-chart-options').remove()
-      this.$container.find('.legend-item').removeClass('active')
-      var otherCharts = this.controller.getOtherCharts(this.chartIndex)
+      this.removePopovers()
+      this.openMoveChart(series, i)
+    }.bind(this))
+  }.bind(this))
```

Comment:
`Array.forEach` supports a second parameter `[this]` so you generally don't need to do `.bind(this)` you can just do `myArray.forEach(function () { ... }, this)`